backend/src/main/java/nl/itvitae/twitbook/user/UserController.java

```diff
@@ -20,19 +24,46 @@
 public class UserController {
 
   private final UserRepository userRepository;
+  private final FollowRepository followRepository;
 
   @GetMapping
   public List<UserDTO> getAll() {
     return userRepository.findAll().stream().map(UserDTO::new).toList();
   }
 
   @GetMapping("by-username/{username}")
-  public ResponseEntity<?> findByUsername(@PathVariable String username) {
+  public ResponseEntity<?> findByUsername(@PathVariable String username,  @AuthenticationPrincipal User authUser) {
     Optional<User> user = userRepository.findByUsernameIgnoreCase(username);
```

Comment:
Now that you added `authUser` in the params, `user` is vague. Rename `authUser` to `user` and `user` to `targetUser` for example